src/main/java/org/repositoryminer/postmining/DefectMetricsPostMiningTask.java

```diff
@@ -0,0 +1,186 @@
+								System.out.println(e.getStackTrace());
+							}
+							
+							Boolean isOpenBefore = false;
+							Boolean isClosedBefore = false;
+							for (Reference reference : tagsList) {
+								Boolean isClosedSameTag = false;
+								if (openDate != null && tags.get(reference.getId()).after(openDate) && !isOpenBefore) {
+									issuesOpen.get(reference.getId()).add(commit);
+									issuesAlive.get(reference.getId()).add(commit);
+									isOpenBefore = true;
+									isClosedSameTag = true;
+								}
+								
+								if (closeDate != null && tags.get(reference.getId()).after(closeDate) && !isClosedBefore) {
+									issuesClosed.get(reference.getId()).add(commit);
+									isClosedBefore = true;
+									if (!isClosedSameTag)
+										issuesAlive.get(reference.getId()).add(commit);
+								}								
+							}
+						}
+					}
+				}				
+			}			
+		}
+	
+		//Calculate occurrences of classes in commits related with issues
+		HashMap<String, HashMap<String, Integer>> classIssuesOpen = new HashMap<String, HashMap<String, Integer>>();
+		HashMap<String, HashMap<String, Integer>> classIssuesClosed = new HashMap<String, HashMap<String, Integer>>();
+		HashMap<String, HashMap<String, Integer>> classIssuesAlive = new HashMap<String, HashMap<String, Integer>>();		
+		for (Reference reference : tagsList) {
+			classIssuesOpen.put(reference.getId(), calculateClassOccurrences(issuesOpen.get(reference.getId())));
+			classIssuesClosed.put(reference.getId(), calculateClassOccurrences(issuesClosed.get(reference.getId())));
+			classIssuesAlive.put(reference.getId(),calculateClassOccurrences(issuesAlive.get(reference.getId())));			
+		}
+		
+		//Save metrics
+		DefectMetricsDocumentHandler dmHandler = new DefectMetricsDocumentHandler();
+		for (Reference reference : tagsList) {
+			Document newDocument = new Document("version",reference.getName());
+			newDocument.append("issues_open", prepareDocument(classIssuesOpen.get(reference.getId())));			
+			newDocument.append("issues_closed", prepareDocument(classIssuesClosed.get(reference.getId())));			
+			newDocument.append("issues_alive", prepareDocument(classIssuesAlive.get(reference.getId())));
+			dmHandler.insert(newDocument);			
+		}		
+	}
+	
+	private List<Document> prepareDocument(HashMap<String, Integer> classList) {
+		List<Document> iOpen = new ArrayList<Document>();
+		
+		for (String c : classList.keySet()) {
+			Document doc = new Document("file", c).append("count", classList.get(c));
+			iOpen.add(doc);
+		}
+		return iOpen;
+	}
+
+	private HashMap<String, Integer> calculateClassOccurrences(List<Commit> commits) {
+		HashMap<String, Integer> classOccurrences = new HashMap<String, Integer>();
+		if (commits.size() > 0) {
+			for (Commit commit : commits) {
+				List<Diff> diffs = commit.getDiffs();
+				if (diffs != null && !diffs.isEmpty()) {
+					for (Diff diff : diffs) {
+						String className = diff.getPath().split("/")[diff.getPath().split("/").length-1];

```

Comment:
I think you should extract `diff.getPath().split("/")` into avariable to avoid multiple method calls as well as to check to avoid null pointer or index out of bound exceptions